src/features/activity/ActivityListItem.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo } from 'react'
+import { ConfirmedSignatureInfo } from '@solana/web3.js'
+import { useTranslation } from 'react-i18next'
+import { EnrichedTransaction } from '../../types/solana'
+import Text from '../../components/Text'
+import Box from '../../components/Box'
+import TouchableOpacityBox, {
+  TouchableOpacityBoxProps,
+} from '../../components/TouchableOpacityBox'
+import { ellipsizeAddress, solAddressIsValid } from '../../utils/accountUtils'
+import CheckmarkFilled from '../../assets/images/checkmarkFill.svg'
+import Error from '../../assets/images/error.svg'
+import { useColors } from '../../theme/themeHooks'
+
+export type ActivityListItemProps = {
+  transaction: EnrichedTransaction | ConfirmedSignatureInfo
+  hasDivider?: boolean
+} & TouchableOpacityBoxProps
+
+const ActivityListItem = ({
+  transaction,
+  hasDivider,
+  ...rest
+}: ActivityListItemProps) => {
+  const colors = useColors()
+  const { t, i18n } = useTranslation()
+
+  const title = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    // Check if transaction is a confirmed signature
+    if (confirmedSig.err || !enrichedTx.type) {
+      return ellipsizeAddress(confirmedSig.signature.toString())
+    }
+
+    const txKey = `activityScreen.enrichedTransactionTypes.${enrichedTx.type}`
+
+    return i18n.exists(txKey)
+      ? t(txKey)
+      : t('activityScreen.enrichedTransactionTypes.UNKNOWN')
+  }, [i18n, t, transaction])
+
+  const subtitle = useMemo(() => {
```

Comment:
Some of the presentational logic in this component seems to be repeated from `ActivityDetailsScreen`. I wonder if we could create a hook or something so we don't have to repeat ourselves.

Reply:
I looked into this and I don't think its worth at this point as the only thing in common is the description